osu.Game/Screens/Edit/Compose/Components/EditorBlueprintContainer.cs

```diff
@@ -63,6 +63,18 @@ protected override void LoadComplete()
             }
         }
 
+        protected override void BeginChange()
+        {
+            // If the editor beatmap has a change handler then this will automatically call BeginChange() on it.
+            EditorBeatmap.BeginChange();
+        }
+
+        protected override void EndChange()
+        {
+            // If the editor beatmap has a change handler then this will automatically call EndChange() on it.
+            EditorBeatmap.EndChange();
+        }
```

Comment:
These two methods are locally forwarded to the editor beatmap in like three places in this diff. What's the reasoning there? Is there a way to not have this?

Reply:
I have these override methods because the base class `BlueprintContainer` wants to indicate the start and end of a change but the base class is used for both beatmap editor and skin editor, and only the beatmap version wants to use the `EditorBeatmap` exclusively. The skin editor still resolves the `IEditorChangeHandler` directly because it obviously doesn't use a `EditorBeatmap`.

I guess I can improve this code by just resolving `IEditorChangeHandler` in the `SkinBlueprintContainer` and resolve none in `BlueprintContainer`. Same thing for the `SelectionHandler` base class.

Reply:
Would caching `EditorBeatmap` as `IEditorChangeHandler` not be a solution for this? I'd much prefer that.

Reply:
It would, but why make it so ambiguous as to what you are notifying about changes? The `EditorBlueprintContainer` would have both a `EditorBeatmap` and a `IEditorChangeHandler`. I prefer having it clear that we are only notifying the `EditorBeatmap` for beatmap changes.